gzip.go

```diff
@@ -166,7 +166,7 @@ func (w *GzipResponseWriter) startGzip() error {
 	// Initialize and flush the buffer into the gzip response if there are any bytes.
 	// If there aren't any, we shouldn't initialize it yet because on Close it will
 	// write the gzip header even if nothing was ever written.
-	if len(w.buf) > 0 {
+	if len(w.buf) >= w.minSize {
```

Comment:
This seems to introduce another issue found in the failing test case due to the reason previously mentioned. This is preventing `w.init()` from being called. Also, the following conditional check `len(w.buf) >= w.minSize` is already made in the `write` function.